cmd/lncli/commands.go

```diff
@@ -2171,10 +2171,42 @@ var listPaymentsCommand = cli.Command{
 	Name:     "listpayments",
 	Category: "Payments",
 	Usage:    "List all outgoing payments.",
+	Description: "This command enables the retrieval of payments stored " +
+		"in the database. Pagination is supported by the usage of " +
+		"index_offset in combination with the paginate_forwards flag. " +
+		"Reversed pagination is enabled by default to receive " +
+		"current payments first. Pagination can be resumed by using " +
+		"the returned last_index_offset (for forwards order), or " +
+		"first_index_offset (for reversed order) as the offset_index. ",
 	Flags: []cli.Flag{
 		cli.BoolFlag{
-			Name:  "include_incomplete",
-			Usage: "if set to true, payments still in flight (or failed) will be returned as well",
+			Name: "include_incomplete",
+			Usage: "if set to true, payments still in flight (or " +
+				"failed) will be returned as well, keeping" +
+				"indices for payments the same as without " +
+				"the flag",
+		},
+		cli.UintFlag{
+			Name: "index_offset",
+			Usage: "The index of a payment that will be used as " +
+				"either the start (in forwards mode) or end " +
+				"(in reverse mode) of a query to determine " +
+				"which payments should be returned in the " +
+				"response, where the index_offset is " +
+				"excluded. If index_offset is set to zero in " +
+				"reversed mode, the query will end with the " +
+				"last payment made.",
+		},
+		cli.UintFlag{
+			Name: "max_payments",
+			Usage: "the max number of payments to return, by " +
+				"default, all completed payments are returned",
+		},
+		cli.BoolFlag{
```

Comment:
Is the default to paginate backwards? I think previously it would always be forwards.

Reply:
Yes, backward pagination is the new default, but it doesn't make a difference compared to the previous output of the command, as before and after the PR all the (completed) payments are returned by default and the order is always oldest to newest payments. In the future when the command is more effective, we could limit the number of returned payments. The motivation as in `listinvoices` and `fwdinghistory` is to be able to quickly look up newest events.